packages/tasks_panel/src/utils/task-serializer.ts

```diff
@@ -1,5 +1,70 @@
 import { ConfiguredTask } from "@sap_oss/task_contrib_types";
+import { randomBytes } from "crypto";
+import escapeStringRegexp = require("escape-string-regexp");
+import { filter, map } from "lodash";
+import { getConfiguredTasksFromCache } from "../../src/services/tasks-provider";
+import { ConfigurationTarget, TaskDefinition, Uri, commands, languages, window, workspace } from "vscode";
 
 export function serializeTask(task: ConfiguredTask): string {
   return JSON.stringify(task);
 }
+
+export function generateUniqueCode(): string {
+  function generateGroup(): string {
+    return randomBytes(2).toString("hex").toUpperCase();
+  }
+  return `${generateGroup()}-${generateGroup()}`;
+}
+
+export function getUniqueTaskLabel(label: string): string {
+  const existingLabels = map(getConfiguredTasksFromCache(), "label");
+  // tasks created from auto detected tasks templates multiple times
+  // will receive name: "<task_name> (<index>)"
+  // where index is the next free number, starting from 2
+  const fixedTaskLabel = escapeStringRegexp(label);
+  const taskRegex = new RegExp(`^${fixedTaskLabel}( [(](\\d)*[)])$`);
```

Comment:
eval named capture groups for increased readability
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Named_capturing_group